packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
       }
     }
 
-    final bool currentDirectory = argResults['current-directory'] && (argResults.wasParsed('current-directory') || dartFiles.isEmpty);
     final bool currentPackage = argResults['current-package'] && (argResults.wasParsed('current-package') || dartFiles.isEmpty);
     final bool flutterRepo = argResults['flutter-repo'] || inRepo(argResults.rest);
 
-    //TODO (pq): revisit package and directory defaults
-
-    if (currentDirectory  && !flutterRepo) {
-      // ./*.dart
-      final Directory currentDirectory = fs.directory('.');
-      bool foundOne = false;
-      for (FileSystemEntity entry in currentDirectory.listSync()) {
-        if (isDartFile(entry)) {
-          dartFiles.add(entry);
-          foundOne = true;
+    // Use dartanalyzer directly except when analyzing the Flutter repository.
+    // Analyzing the repository requires a more complex report than dartanalyzer
+    // currently supports (e.g. missing member dartdoc summary).
+    // TODO(danrubel) enhance dartanalyzer to provide this type of summary
+    if (!flutterRepo) {
+      final List<String> arguments = <String>[];
+      arguments.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (arguments.length < 1 || currentPackage) {
+        final Directory projectDirectory = await projectDirectoryContaining(fs.currentDirectory.absolute);
+        if (projectDirectory != null)
+          arguments.add(projectDirectory.path);
+      } else {
+        String currentDirectoryPath = fs.currentDirectory.path;
+        if (!currentDirectoryPath.endsWith(fs.path.separator))
+          currentDirectoryPath += fs.path.separator;
+
+        // If the files being analyzed are outside of the current directory hierarchy
+        // then dartanalyzer does not yet know how to find the ".packages" file.
+        // In this situation, use the first file as a starting point
+        // to search for a "pubspec.yaml" and ".packages" file.
+        // TODO(danrubel): fix dartanalyzer to find the .packages file
+        if (!arguments[0].startsWith(currentDirectoryPath)) {
```

Comment:
I think we should explicitly check that all the files would find the same `.packages` (or all would find no `.packages`?), and exit with a message saying that we don't yet support checking multiple packages at once if you're not also checking the flutter repo. I've definitely run into the case where I've tried to use `flutter analyze` to analyze two packages at once (an app I was working on and the support library I was writing for the app), and it would be massively confusing if that did something different depending on the order in which the paths were listed.